src/olympia/api/throttling.py

```diff
@@ -28,3 +46,32 @@ def parse_rate(self, rate):
         multipler = int(period_num) if period_num else 1
         duration = {'s': 1, 'm': 60, 'h': 3600, 'd': 86400}[period] * multipler
         return (num_requests, duration)
+
+
+class GranularIPRateThrottle(GranularUserRateThrottle):
+    """
+    Throttling class that works like DRF's AnonRateThrottle but is always
+    applied, even to authenticated requests, supports granular rates like
+    1/5second, and can be deactivated through a API_THROTTLING django setting.
+
+    Its scope defaults to `anon` to follow DRF's default behaviour regarding
+    IP throttling but in most cases we'll want the child class to override that
+    and add a custom rate.
+    """
+    scope = 'anon'
+
+    def get_cache_key(self, request, view):
+        return self.cache_format % {
+            'scope': self.scope,
+            'ident': self.get_ident(request)  # This will get the IP.
+        }
+
+
+class ThrottleOnlyUnsafeMethodsMixin():
```

Comment:
I can't help thinking this should be a wrapper class, or a class variable (e.g. `unsafe_methods_only = False` by default, that can be overridden by subclasses) in `GranularUserRateThrottle`.  But personal opinion and not a blocker given the issue priority.

Reply:
Yeah I can do that in a follow-up. Was trying not to impact other existing views for now.